packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,116 @@
+/*
+ * Utility that wraps around ethereumjs-abi, providing additional validation
+ * on passed type/value pair parameters.
+ */
+
 import abi from 'ethereumjs-abi';
+import * as util from 'ethereumjs-util';
 import BN from 'bignumber.js';
+import _ from 'lodash';
 
-function formatValue(value: any): string {
-  if (typeof(value) === 'number' || BN.isBigNumber(value)) return value.toString();
-  else if (typeof(value) === 'string' && value.match(/\d+(\.\d+)?e(\+)?\d+/)) return (new BN(value)).toString(10);
-  else return value;
-}
+const ERROR_MESSAGE_PREFIX = 'Encoding error';
+const ERROR_MESSAGE = (type: string, value: any) => `${ERROR_MESSAGE_PREFIX} for type ${type} and value ${value.toString()}`;
 
-export default function encodeCall(name: string, args: string[] = [], rawValues: any[] = []): string {
-  const values: string[] = rawValues.map(formatValue);
-  const methodId: string = abi.methodID(name, args).toString('hex');
-  const params: Buffer = abi.rawEncode(args, values).toString('hex');
+export default function encodeCall(name: string, types: string[] = [], values: any[] = []): string {
+  const methodId = abi.methodID(name, types).toString('hex');
+  const params = abi.rawEncode(types, values).toString('hex');
   return '0x' + methodId + params;
 }
+
+export function parseCallValues(types: string[] = [], rawValues: any[] = []): any[] | never {
```

Comment:
If we are now using parse just to handle input strings, shouldn't rawValues be a string[]? Which use case do we want to cover by accepting a `bool` value for `parseBool`, for instance?

Reply:
Actually, encodeCall should also have values typed to any[] and not string[]. That's because ethereumjs-abi is not expecting values to be just strings. That makes me hesitate a bit about removing validation from programmatic usage...
In the case of parseCallValues, most of the parsing functions are prepared to receive non string values, such as parseNumber, which can receive number types as well as big number types, or the case of parseBytes, that can receive Buffers.

Reply:
> Actually, encodeCall should also have values typed to any[] and not string[]. That's because ethereumjs-abi is not expecting values to be just strings. 

Yes, I'd expect encodeCall to work with any[], not string[], since it should receive the correct type (or something that can be easily coercerd to) for each value.

> In the case of parseCallValues, most of the parsing functions are prepared to receive non string values, such as parseNumber, which can receive number types as well as big number types, or the case of parseBytes, that can receive Buffers.

I see, but my point is: if we are *only* using parseCallValues to parse strings, why support scenarios that we never use? I'd rather clear that code out if it's never used.